application/server/configuration.go

```diff
@@ -127,6 +132,29 @@ func writeSettings(settings lsp.Settings, initialize bool) {
 	updateSnykCodeQuality(settings)
 	updateRuntimeInfo(settings)
 	updateAutoScan(settings)
+	updateAuthenticationMethod(settings)
+}
+
+func updateAuthenticationMethod(settings lsp.Settings) {
+	c := config.CurrentConfig()
+	c.SetAuthenticationMethod(settings.AuthenticationMethod)
+	if config.CurrentConfig().GetAuthenticationMethod() == lsp.OAuthAuthentication {
+		engine := di.Engine()
+		c := engine.GetConfiguration()
+		c.Set(configuration.OAUTH_AUTH_ENABLED, true)
+		c.Set(configuration.API_URL, config.CurrentConfig().SnykApi())
```

Comment:
it might make sense to do this somewhere else, since we need this with and without oauth

Reply:
yep, will come with config sync